### PR TITLE
Allow RTPS writers to be constructed with custom pools [9432]

### PR DESCRIPTION
**Pull request by MiguelCompany**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2728/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/111/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2514/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2739/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2732/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/114/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2517/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2743/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2733/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/115/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2518/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2744/)

File: src/cpp/rtps/writer/RTPSWriter.cpp [hunk 1]
```diff
@@ -18,40 +18,111 @@
...
+                    {
+                        payload_pool->release_payload(*item);
+                    }
+                };
+
+        return std::make_shared<CacheChangePool>(pool_config, init_cache);
+    }
+
+    return std::make_shared<CacheChangePool>(pool_config);
+}
```
Comment (IkerLuengo) [inline src/cpp/rtps/writer/RTPSWriter.cpp:58]: We could put this implementation in `RTPSWriter::create_change_pool` and avoid an extra function.

**Review by IkerLuengo — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/writer/RTPSWriter.cpp (hunk 1 above)
Comment (MiguelCompany) [reply src/cpp/rtps/writer/RTPSWriter.cpp]: I refactored this with a new method on BasicPayloadPool returning both pools.

**Review by IkerLuengo — APPROVED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2736/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/118/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2521/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2747/)

**Comment by MiguelCompany** [comment]
Failure summary:
* uncrustify: lambda not treated correctly
* Mac: Would be fixed when #1186 is ported to master
* Linux: Fixed by #1411

**Pull request merged**